runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/AbstractWindmillStream.java

```diff
@@ -176,6 +176,7 @@ protected final void startStream() {
           onNewStream();
           if (clientClosed.get()) {
             close();
+            streamRegistry.remove(this);
```

Comment:
it seems like close should trigger onStreamFinished which would do this

Reply:
removed

Reply:
looks like close() calls requestObserver.complete() but onStreamFinished is called by the responseObserver.complete()?